python/ray/data/_internal/logical/rules/operator_fusion.py

```diff
@@ -204,6 +204,12 @@ def _can_fuse(self, down_op: PhysicalOperator, up_op: PhysicalOperator) -> bool:
         ):
             return False
 
+        # Only fuse if at most one op specifies a `_ray_remote_args_fn`.
```

Comment:
I tend to not fuse if any has the fn. because the fn can return any args that may be incompatible with the other.

Reply:
Yes that's what i was thinking as well. But in the case of `Read->Map` with `ray_remote_args_fn` in the `Map`, then these two operators would not be fused, which would lead to performance drop.

Reply:
do you have a concrete example of how bad the perf drop is? I guess at least for vLLM integration, it's not an issue, right? because the actors are GPU actors, they cannot be fused anyway. 
Despite the perf drop in some cases, I still think, for correctness, we shouldn't fuse them, unless we have a way to tell whether the returned args are compatible with the previous op.